compiler-cli/src/lsp.rs

```diff
@@ -371,31 +378,37 @@ fn error_to_diagnostic(error: &Error) -> Option<PublishDiagnosticsParams> {
                 message.push_str(&extra);
             }
             message.push('\n');
-            let diagnostic = Diagnostic {
-                range: src_span_to_lsp_range(location, line_numbers),
-                severity: Some(DiagnosticSeverity::ERROR),
-                code: None,
-                code_description: None,
-                source: None,
+
+            let diagnostic_params = new_diagnostic(
+                src.to_string(),
+                path.to_path_buf(),
+                error.location,
+                DiagnosticSeverity::ERROR,
                 message,
-                related_information: None,
-                tags: None,
-                data: None,
-            };
-            let path = path.canonicalize().unwrap();
-            let mut file: String = "file://".into();
-            file.push_str(&path.as_os_str().to_string_lossy());
-            let uri = Url::parse(&file).unwrap();
-            let diagnostic_params = PublishDiagnosticsParams {
-                uri,
-                diagnostics: vec![diagnostic],
-                version: None,
-            };
+            );
             Some(diagnostic_params)
         }
 
-        Error::Type { .. }
-        | Error::UnknownImport { .. }
+        Error::Type { error, src, path } => {
```

Comment:
Rather than pattern matching on each error variant we want to convert the error to a diagnostic no matter what it is, so we can remove the entire `match` expression